moxin-frontend/src/chat/chat_panel.rs

```diff
@@ -254,8 +248,8 @@ pub struct ChatPanel {
     #[rust]
     auto_scroll_cancellable: bool,
 
-    #[rust(true)]
-    prompt_enabled: bool,
+    #[rust]
+    is_chat_streaming: bool,
```

Comment:
(This may be out of the scope of the PR, we can discuss it later or smth, just a thought from a quick glance)

These 4 flags are starting to make the rest of the widget a bit messy and hard to follow, does it make sense to go with a sort of state machine approach here? maybe a full FSM is a bit of an overkill, but we could start encapsulating the state in an enum and have a couple of functions in the struct to handle stuff like the scrolling and state transitioning. Again, not wanting to make stuff overcomplicated here so lmk if im overthinking. 
```rust
enum ChatPanelState {
    Loading,
    Ready { auto_scroll_pending: bool, auto_scroll_cancellable: bool },
    Streaming,
}
```